### PR TITLE
Feat: Fork a Repo

### PR DESCRIPTION
**Pull request by li-xin-yi**
image

Make a copy of the current repo and navigate to the copied repo (untitled) in-place. 

Because the only difference is the repo id (URL), so it looks exactly the same after clicking on the button, just keep an eye on the address in your browser.

### CODE REVIEW

**Comment by lihebi** [comment]
I think we'd call it "make a copy" instead of "fork" because we're not tracking the upstream repo like what GitHub did. This "make-a-copy" feature is good enough, as we don't have a proper VCS to support the real fork now.

Also, it would be nice to name the new repo "copy-of-XXX" instead of leaving it untitled, where XXX could be:
* if the original repo has the title, use that title
* otherwise, use the original repo's ID

**Comment by li-xin-yi** [comment]
> I think we'd call it "make a copy" instead of "fork" because we're not tracking the upstream repo like what GitHub did.

I agree.

Besides, open a new tab or navigate to the copied repo in place, which method do you prefer?

**Comment by lihebi** [comment]
Good idea, I’d prefer opening a new tab.

File: ui/src/pages/repo.tsx [hunk 1]
```diff
@@ -180,6 +193,19 @@ function RepoWrapper({ children, id }) {
               Share
             </Button>
           }
+          forkButton={
+            <Button
+              endIcon={<ContentCopyIcon />}
+              onClick={async () => {
+                const result = await copyRepo();
```
Comment (lihebi) [inline ui/src/pages/repo.tsx:200]: This is interesting; I didn't know we could await for the mutation function. Looks like we indeed can.

File: api/src/resolver_repo.ts [hunk 2]
```diff
@@ -497,6 +497,68 @@ async function deletePods(_, { ids }: { ids: string[] }, { userId }) {
...
+      ...pod,
+      id: idMap.get(pod.id),
+      parent: pod.parent ? { id: idMap.get(pod.parent.id) } : undefined,
+      repoId: id,
+      parentId: pod.parentId ? idMap.get(pod.parentId) : undefined,
+    };
+  });
+
+  // Add all nodes without parent/child relationship to the new repo.
+  // TODO: it updates the parent/child relationship automatically somehow,maybe because the parentId? Try to figure out why, then refactor addPods method.
```
Comment (lihebi) [inline api/src/resolver_repo.ts:549]: Two comments:
1. The parent/children relationship is a **single** relation (named the `PARENT` relation in the schema). Thus, updating one should automatically update another. We should probably only update the `parent` field and not touch the `children` field at all.
2. Yep, providing the `parentId` field seems to be equivalent of using `{connect: {parent: {id}}}` Prisma clause.

**Review by lihebi — APPROVED** [review]

**Pull request merged**